pallene/coder.lua

```diff
@@ -479,13 +479,25 @@ function Coder:pallene_entry_point_definition(f_id)
     }))
 end
 
-function Coder:call_pallene_function(dsts, has_upvalue, f_id, base, xs)
+function Coder:call_pallene_function(dsts, f_id, base, xs, cclosure)
+
+    local func       = self.module.functions[f_id]
+    local n_upvalues = #func.captured_vars
 
     local args = {}
     table.insert(args, "L")
     table.insert(args, base)
     table.insert(args, "G")
-    table.insert(args, has_upvalue and "func->upvalue" or "NULL")
+
+    local upvals
```

Comment:
If I were not familiar with this optimization, I would have guessed that the reason for the NULL is because the upvalue array doesn't exist when there are no captured variables. Which is not the real reason...

Maybe we should add a comment saying that this NULL thing an optimization.